### DISCUSSION TITLE
Docker nginx attachment - getting 413

### DISCUSSION THREAD

**Discussion by drissical** [kanboard/kanboard#5198] [Installation & Configuration]
### Actual behaviour
Using the docker image I am not able to attach anything over 1M in size.  I am able to upload files that are ~950K, I was able to zip a failing 1M file to just under 1M and get it uploaded like that.  I checked the size settings per the documentation, and they both look ok.
Logging in to cli via `docker exec`
```
bash-5.0# php -i | grep -e "filesize" -e "max_size"
post_max_size => 32M => 32M
upload_max_filesize => 32M => 32M
```
```
bash-5.0# grep client nginx/*
nginx/nginx.conf:        client_max_body_size      32m;
```
(I am not sure how to check the running config of nginx like with php so I am assuming that the setting is either incorrect, ignored, overridden)

### Expected behaviour
The PHP settings and Nginx settings look like they are set to 32M already. I would expect to be able to upload an attachment up to 32M.

### Steps to reproduce

1. Docker run the latest image
2. Login to the site
3. Create a project
4. Create a task
5. Attach a file over 1M

### Screenshots
![image](https://user-images.githubusercontent.com/8355446/108882225-0187ab00-75b9-11eb-883a-57ca275d4df0.png)
![image](https://user-images.githubusercontent.com/8355446/108882380-2da32c00-75b9-11eb-87cd-6ca3192dad5e.png)

### Logs
Logs actually do not show any errors (i dont think it is generating an error, just a legitimate error code for the user). I stopped and started nginx without the daemon mode settings (`nginx -g "daemon off;"`) so that the logs would go directly to the console and I only seen some PHP constant already defined errors which is fine. They are related to the database and my guess is that these just exist in the docker verison.

### Configuration
![image](https://user-images.githubusercontent.com/8355446/108883673-97700580-75ba-11eb-9b5c-1e806d73897c.png)

- Kanboard version: v1.2.18
- Database type and version: 10.4.17-MariaDB-1:10.4.17+maria~stretch
- PHP version: 7.3.25
- OS: Linux 4.19.0-13-amd64
- Browser: (Using the Chromium Edge) Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4412.0 Safari/537.36 Edg/90.0.796.0

**Comment by drissical** [comment]
I just realized that the `php -i` might not pull the same settings as it is cli vs nginx/fpm, just incase I created a phpinfo and threw that in the web dir.
![image](https://user-images.githubusercontent.com/8355446/108884836-e5393d80-75bb-11eb-9b48-5faa9ed4850f.png)

**Comment by fguillot** [comment]
I just tested with the Docker image v1.2.18 and I am able to upload files up to the limit of 32MB.

However, I can see on your screenshot that you have Nginx 1.19.3, but the Docker image comes with Nginx 1.18.0 which makes me think that you have a reverse proxy in front of the Kanboard container. This Nginx has probably different settings that the one bundled into the Docker image.

**Comment by drissical** [comment]
Ah I did not even think about that.  Thanks for catching that, going to go down that route now!

Closing this issue for now.
